lumicks/pylake/fitting/model.py

```diff
@@ -88,14 +88,19 @@ def __init__(
 
             fit.plot("my data", "k--")  # Plot the fitted model
         """
-        assert isinstance(name, str), "First argument must be a model name."
-        assert isinstance(model_function, types.FunctionType), "Model must be a callable."
+        if not isinstance(name, str):
+            raise TypeError("First argument must be a model name.")
+
+        if not isinstance(model_function, types.FunctionType):
```

Comment:
you can actually use the built-in `callable()`
https://docs.python.org/3/library/functions.html#callable